pkgs/build-support/singularity-tools/default.nix

```diff
@@ -108,14 +116,14 @@ rec {
 
             # Build /bin and copy across closure
             mkdir -p bin ./${builtins.storeDir}
-            for f in $(cat $layerClosure) ; do
-              cp -ar $f ./$f
-            done
+            while IFS= read -r f; do
+              cp -r "$f" "./$f"
+            done < "$layerClosure"
```

Comment:
> writeClosure produces line-separated outputs?

Yes, so do all the other reference-related build helpers.

> I guess we should mention this briefly either here or in the nix file

Good idea!

Reply:
I guess adding a "line-separated" wouldn't hurt but there's already an example in there that makes it clear. My only concern was "uh is this the right `read` incantation?" while reviewing this diff, because I usually see something cursed like `read -r -d $'\0'` or `read -r -d ""` for `\0`-split inputs...

Reply:
Addressed.